Added documentation for tests:check-direct-dependency-use command, #PG-3272 (#778)
piwik/developer-documentation: docs/5.x/tests-system.md

}
```

## How to write a test to scan for Matomo core dependencies being used directly in your plugins

> With the release of Matomo `v5.1.0`, you can now check the usage of core dependencies in your plugin directly using the `tests:check-direct-dependency-use` command. With the release of Matomo 5, plugins should not use core dependencies directly but instead [prefix](https://developer.matomo.org/guides/migrate-matomo-4-to-5#vendor-proxies) them. You can also write a system test case to test the same by using the sample code below

Reviewer: ```suggestion
With the release of Matomo `5.1.0`, you can now easily check if your plugin is using any core dependencies by running the `tests:check-direct-dependency-use` command. With the release of Matomo 5, plugins should not use core dependencies directly but instead use them via [proxies](https://developer.matomo.org/guides/migrate-matomo-4-to-5#vendor-proxies). To have a check for this in your test suite and run it automatically, you can also write a system test case by using the sample code below:
```

> ```php
> <?php
> 
> use Piwik\Plugins\TestRunner\Commands\CheckDirectDependencyUse;

Reviewer: ```suggestion
```php
use Piwik\Plugins\TestRunner\Commands\CheckDirectDependencyUse;
```

Reviewer: I checked other examples and we don't use the opening tag in them.

use Piwik\Tests\Framework\TestCase\SystemTestCase;
use Piwik\Version;
use Symfony\Component\Console\Input\ArrayInput;
use Symfony\Component\Console\Output\NullOutput;

class CheckDirectDependencyUseCommandTest extends SystemTestCase
{
    public function testCommand()
    {

>         if (version_compare(Version::VERSION, '5.0.2', '<=') && !\Piwik\file_exists(PIWIK_INCLUDE_PATH . '/plugins/TestRunner/Commands/CheckDirectDependencyUse.php')) {

Reviewer: Should this test for versions before 5.1.0 or when the file is missing like so?

```suggestion
        if (version_compare(Version::VERSION, '5.1.0', '<') || !\Piwik\file_exists(PIWIK_INCLUDE_PATH . '/plugins/TestRunner/Commands/CheckDirectDependencyUse.php')) {
```

            $this->markTestSkipped('tests:check-direct-dependency-use is not available in this version');

>         }
>         $pluginName = '{YOUR_PLUGIN_NAME}';
>         $console = new \Piwik\Console(self::$fixture->piwikEnvironment);
>         $checkDirectDependencyUse = new CheckDirectDependencyUse();
>         $console->addCommands([$checkDirectDependencyUse]);
>         $command = $console->find('tests:check-direct-dependency-use');
>         $arguments = array(
>             'command'    => 'tests:check-direct-dependency-use',
>             '--plugin' => $pluginName,
>             '--grep-vendor'
>         );
>         $inputObject = new ArrayInput($arguments);
>         $command->run($inputObject, new NullOutput());

Reviewer: Just a bit of formatting tweak for easier reading of the code.

```suggestion
        }
        
        $pluginName = '{YOUR_PLUGIN_NAME}';
        
        $checkDirectDependencyUse = new CheckDirectDependencyUse();
        $console = new \Piwik\Console(self::$fixture->piwikEnvironment);
        $console->addCommands([$checkDirectDependencyUse]);
        
        $command = $console->find('tests:check-direct-dependency-use');
        $arguments = [
            'command'  => 'tests:check-direct-dependency-use',
            '--plugin' => $pluginName,
            '--grep-vendor',
        ];
        $inputObject = new ArrayInput($arguments);
        $command->run($inputObject, new NullOutput());
```
